java-frontend/src/main/java/org/sonar/java/SonarComponents.java

```diff
@@ -152,6 +153,22 @@ public NewHighlighting highlightableFor(InputFile inputFile) {
     return javaTestClasspath.getElements();
   }
 
+  public List<File> getJspClasspath() {
+    List<File> jspClasspath = new ArrayList<>();
+    jspClasspath.add(findPluginJar());
+    jspClasspath.addAll(javaClasspath.getElements());
```

Comment:
this would probably be more robust to call for getJavaClasspath rather than relying on order of callers for javaclasspath to be initialized.